File: specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-07-02/examples-js/diskExamples/Disk_ListBySubscription.js
const { ComputeManagementClient } = require("@azure/arm-compute");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Lists all the disks under a subscription.
 *
 * @summary Lists all the disks under a subscription.
 * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-07-02/examples/diskExamples/Disk_ListBySubscription.json
 */
async function listAllManagedDisksInASubscription() {
  const subscriptionId = "{subscription-id}";
  const credential = new DefaultAzureCredential();
  const client = new ComputeManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.disks.list()) {
    resArray.push(item);
  }
  console.log(resArray);
}

listAllManagedDisksInASubscription().catch(console.error);

File: specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-07-02/examples-java/diskExamples/Disk_ListBySubscription.java
import com.azure.core.util.Context;

/** Samples for Disks List. */
public final class Main {
    /*
     * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-07-02/examples/diskExamples/Disk_ListBySubscription.json
     */
    /**
     * Sample code: List all managed disks in a subscription.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void listAllManagedDisksInASubscription(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.virtualMachines().manager().serviceClient().getDisks().list(Context.NONE);
    }
}
